Kudu.Services/Filters/RequestThrottleActionFilterAttribute.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Web.Http.Controllers;
+using System.Web.Http.Filters;
+
+namespace Kudu.Services.Filters
+{
+    public sealed class RequestThrottleActionFilterAttribute : ActionFilterAttribute
+    {
+        int _concurrency = 0;
+
+        public int MaxConcurrency { get; set; } = 5;
+
+        public override void OnActionExecuting(HttpActionContext actionContext)
+        {
+            // If we've already reached the concurrency limit, return 429 (too many requests)
+            // and ask to retry after 5 seconds
+            if (_concurrency >= MaxConcurrency)
+            {
+                actionContext.Response = new HttpResponseMessage((HttpStatusCode)429);
```

Comment:
In the future, we will need to inspect how many sites fall into this.   I don't know if 429  from IISLog is obvious enough.   We might consider ETW tracing `once` when we get it here and reset that state when it is back to normal.

```csharp
TraceServices.CurrentRequestTracer?.Trace(...);
```